### PR TITLE
GCC 8 fixes

### PR DESCRIPTION
**Pull request by janisozaur**
This fixes https://github.com/OpenRCT2/OpenRCT2/issues/6981

### CODE REVIEW

File: src/openrct2-ui/windows/GuestList.cpp [hunk 1]
```diff
@@ -856,7 +856,8 @@ static void window_guest_list_textinput(rct_window *w, rct_widgetindex widgetInd
 {
     if (text != nullptr && text[0] != '\0')
     {
-        strncpy(_window_guest_list_filter_name, text, sizeof(_window_guest_list_filter_name));
+        strncpy(_window_guest_list_filter_name, text, sizeof(_window_guest_list_filter_name) - 1);
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/GuestList.cpp]: Don't we have safe_strncpy to take care of null pointers?

File: src/openrct2/interface/Screenshot.cpp [hunk 2]
```diff
@@ -126,7 +127,7 @@ static sint32 screenshot_get_next_path(char *path, size_t size)
     sint32 i;
     for (i = 1; i < 1000; i++) {
         // Glue together path and filename
-        snprintf(path, size, "%s%s %d-%02d-%02d %02d-%02d-%02d (%d).png", screenshotPath, park_name, currentDate.year, currentDate.month, currentDate.day, currentTime.hour, currentTime.minute, currentTime.second, i);
+        snprintf(fileNameCh, leftBytes, "%s %d-%02d-%02d %02d-%02d-%02d (%d).png", park_name, currentDate.year, currentDate.month, currentDate.day, currentTime.hour, currentTime.minute, currentTime.second, i);
```
Comment (Gymnasiast) [inline src/openrct2/interface/Screenshot.cpp]: Why are you taking out `screenshotPath`?

Comment (janisozaur) [reply src/openrct2/interface/Screenshot.cpp]: it's already set in `fileNameCh`

**Comment by janisozaur** [comment]
`safe_strcpy` + rebased

**Pull request merged**